### PR TITLE
freeswitch: patch CVE-2023-5217 and CVE-2023-44488

### PR DESCRIPTION
**Pull request by MikaelFangel**
## Description of changes

Applies the security patch because no upstream release has been made yet, and it looks like they have a slow release cycle. To see the pr, used for the patch, look [here](https://github.com/signalwire/freeswitch/pull/2266)

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/sip/freeswitch/default.nix [hunk 1]
```diff
@@ -110,6 +126,15 @@ stdenv.mkDerivation rec {
     done
   '';
 
+  ## TODO Validate with the next upstream release
+  pathces = [
+    (fetchpatch {
+       name = "CVE-2023-5217.patch";
+       url = "https://github.com/signalwire/freeswitch/commit/f1fb05214e4f427dcf922f531431ab649cf0622b.patch";
+       sha256 = "sha256-6GMebE6O2EBx60NE2LSRVljaiLm9T4zTrkIpwGvaB08=";
```
Comment (eclairevoyant) [inline pkgs/servers/sip/freeswitch/default.nix:134]:
```suggestion
       hash = "sha256-6GMebE6O2EBx60NE2LSRVljaiLm9T4zTrkIpwGvaB08=";
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/servers/sip/freeswitch/default.nix (hunk 1 above)
Comment (MikaelFangel) [reply pkgs/servers/sip/freeswitch/default.nix]: Thanks!

**Review by eclairevoyant — APPROVED** [review]

File: pkgs/servers/sip/freeswitch/default.nix [hunk 2]
```diff
@@ -110,6 +126,15 @@ stdenv.mkDerivation rec {
     done
   '';
 
+  ## TODO Validate with the next upstream release
+  pathces = [
```
Comment (LeSuisse) [inline pkgs/servers/sip/freeswitch/default.nix:130]:
```suggestion
  patches = [
```

File: pkgs/servers/sip/freeswitch/default.nix [hunk 3]
```diff
@@ -110,6 +126,15 @@ stdenv.mkDerivation rec {
...
+    (fetchpatch {
+       name = "CVE-2023-5217.patch";
+       url = "https://github.com/signalwire/freeswitch/commit/f1fb05214e4f427dcf922f531431ab649cf0622b.patch";
```
Comment (LeSuisse) [inline pkgs/servers/sip/freeswitch/default.nix:133]: I think you want https://github.com/signalwire/freeswitch/commit/6f9e72c585265d8def8a613b36cd4f524c201980 for CVE-2023-5217.

We also want this patch but it is for CVE-2023-44488.

**Review by LeSuisse — CHANGES_REQUESTED** [review]

File: pkgs/servers/sip/freeswitch/default.nix (hunk 3 above)
Comment (MikaelFangel) [reply pkgs/servers/sip/freeswitch/default.nix:133]: Oh, I see! I'll fix it right away. Just waiting to see if it builds (:

Comment (MikaelFangel) [reply pkgs/servers/sip/freeswitch/default.nix:133]: @LeSuisse Have added both patches now and tested that it builds.

File: pkgs/servers/sip/freeswitch/default.nix (hunk 2 above)
Comment (LeSuisse) [reply pkgs/servers/sip/freeswitch/default.nix:130]: Unless I'm missing something this change is still needed.

Comment (MikaelFangel) [reply pkgs/servers/sip/freeswitch/default.nix]: You're correct.

**Comment by risicle** [comment]
Well at least they patch their bundled dependencies.

**Comment by risicle** [comment]
`nixpkgs-review` happy, nixos x86_64. `passthru.tests` pass, `pkgsi686Linux` variant builds.

**Review by LeSuisse — APPROVED** [review]
Looks good to me too. Let's merge.

**Pull request merged**